### PR TITLE
Add Web Sequence Diagrams filter

### PR DESCRIPTION
**Pull request by jamespjh**
Add a filter for https://www.websequencediagrams.com/

### CODE REVIEW

**Comment by jamespjh** [comment]
Not yet ready for pull, needs test and doc.
Will comment again when ready.

**Comment by jamespjh** [comment]
@ananelson please Code Review.

File: dexy/filters/websequence.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        line = response.readline()
+        response.close()
+
+        expr = re.compile("(\?(img|pdf|png|svg)=[a-zA-Z0-9]+)")
+        match = expr.search(line)
+
+        if match is None:
+            self.log_error("Problem fetching from WebSequenceDiagrams")
+            raise UserFeedback("Invalid response from server.")
+            return
```
Comment (ananelson) [inline dexy/filters/websequence.py]: no point putting 'return' after raise since it'll never be run

File: dexy/filters/websequence.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+from ..exceptions import UserFeedback, InternalDexyProblem
```
Comment (ananelson) [inline dexy/filters/websequence.py]: for consistency, I use "dexy.exceptions" throughout, although if there's a good case to make for using .. instead I'll consider switching everywhere.

File: dexy/filters/websequence.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    def process(self):
+        request = {}
+        settings = self.setting_values()
+        request["message"] = self.input_data
+        request["style"] = settings["style"]
+        request["apiVersion"] = settings["version"]
+        key = None
+        try:
+            key = settings["key"] or self.read_param("key")
+        except Exception:   # Need to modify filters.api.read_param to
```
Comment (ananelson) [inline dexy/filters/websequence.py]: as discussed in person, please go ahead and make this modification to filters.api.read_param

**Pull request merged**